src/views/AverageBlockTimeChart.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <div class="container">
+    <div class="page-title">
+      <BackButton :current-router="router" :text="'Charts & Stats'" />
+      <h2 class="view-title">AverageBlockTimeChart</h2>
+    </div>
+
+    <div class="sort-wrapper">
+      <span>
+        Block Time in<br />
+        Seconds
+      </span>
+
+      <VuePicker
+        class="app-form__field-input app-filter app-filter--coin"
+        name="filter"
+        v-model="sortingValue"
+        :isDisabled="isLoading"
+      >
+        <template #dropdownInner>
+          <div class="app-filter__dropdown-inner">
+            <VuePickerOption
+              v-for="{ text, value } in sortingDaysForChart"
+              :key="text"
+              :value="value"
+              :text="text"
+            >
+              {{ text }}
+            </VuePickerOption>
+          </div>
+        </template>
+      </VuePicker>
+    </div>
+    <CustomBarChart
+      :data="chartData"
+      :datasetLabel="'Block Time'"
+      :datasetUnit="'sec'"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, onMounted, ref, watch } from 'vue'
+import { Router, useRouter } from 'vue-router'
+import { callers } from '@/api/callers'
+import { handleError } from '@/helpers/errors'
+import { formatDataForCharts } from '@/helpers/customChartHelpers'
+import { sortingDaysForChart } from '@/helpers/helpers'
+import CustomBarChart from '@/components/Charts/CustomBarChart.vue'
+import BackButton from '@/components/BackButton.vue'
+
+export default defineComponent({
+  name: 'AverageBlockTimeChart',
+  components: { CustomBarChart, BackButton },
+  setup() {
+    const router: Router = useRouter()
+    const chartData = ref()
+    const isLoading = ref<boolean>(false)
+    const sortingValue = ref<string>('1')
+
+    const getChartData = async () => {
+      const endDate = new Date()
+      const startDate = new Date()
+      startDate.setDate(startDate.getDate() - Number(sortingValue.value))
+
+      isLoading.value = true
+      try {
+        const { data } = await callers.getAvgTimePerDays(startDate, endDate)
+
+        chartData.value = formatDataForCharts(data)
+      } catch (error) {
+        handleError(error as Error)
+        console.error(error)
+      } finally {
+        isLoading.value = false
+      }
+    }
+
+    watch(sortingValue, async (): Promise<void> => await getChartData())
```

Comment:
maybe call it on vue-picker @input event?

Reply:
there is no input event in this custom component